server/src/main/java/org/opensearch/action/admin/cluster/hahealth/ClusterHaHealthResponse.java

```diff
@@ -0,0 +1,141 @@
+    private static final String AWARENESS_ATTRIBUTE = "AwarenessAttribute";
+
+    private String clusterName;
+    private ClusterHaHealth clusterHaHealth;
+
+    public ClusterHaHealthResponse() {}
+
+    public ClusterHaHealthResponse(StreamInput in) throws IOException {
+        super(in);
+        clusterName = in.readString();
+        clusterHaHealth = new ClusterHaHealth(in);
+    }
+
+    public ClusterHaHealthResponse(ClusterState clusterState, ClusterSettings clusterSettings) {
+        this.clusterName = clusterState.getClusterName().value();
+        this.clusterHaHealth = new ClusterHaHealth(clusterState, clusterSettings);
+    }
+
+    // Constructor used by Unit tests
+    public ClusterHaHealthResponse(
+        String clusterName,
+        int activeShards,
+        int unassignedShards,
+        int initializingShards,
+        int numberOfDataNodes,
+        int numberOfNodes,
+        ClusterHealthStatus clusterHealthStatus
+    ) {
+        this.clusterName = clusterName;
+        this.clusterHaHealth = new ClusterHaHealth(
+            activeShards,
+            unassignedShards,
+            initializingShards,
+            numberOfDataNodes,
+            numberOfNodes,
+            clusterHealthStatus
+        );
+    }
+
+    // Defined for Unit Tests
+    public static ClusterHaHealthResponse readResponse(StreamInput in) throws IOException {
+        return new ClusterHaHealthResponse(in);
+    }
+
+    public String getClusterName() {
+        return clusterName;
+    }
+
+    public int getActiveShards() {
+        return clusterHaHealth.getActiveShards();
+    }
+
+    public int getUnassignedShards() {
+        return clusterHaHealth.getUnassignedShards();
+    }
+
+    public int getInitializingShards() {
+        return clusterHaHealth.getInitializingShards();
+    }
+
+    public int getNumberOfDataNodes() {
+        return clusterHaHealth.getNumberOfDataNodes();
+    }
+
+    public int getNumberOfNodes() {
+        return clusterHaHealth.getNumberOfNodes();
+    }
+
+    public Map<String, Object> getAwarenessAttributeHealth() {
+        return clusterHaHealth.getAwarenessAttributeHealth();
+    }
+
+    public ClusterHealthStatus getStatus() {
+        return clusterHaHealth.getStatus();
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field(CLUSTER_NAME, getClusterName());
+        builder.field(STATUS, getStatus().name().toLowerCase(Locale.ROOT));
+        builder.field(NUMBER_OF_NODES, getNumberOfNodes());
+        builder.field(NUMBER_OF_DATA_NODES, getNumberOfDataNodes());
+        builder.field(ACTIVE_SHARDS, getActiveShards());
+        builder.field(UNASSIGNED_SHARDS, getUnassignedShards());
+        builder.field(INITIALIZING_SHARDS, getInitializingShards());
+        builder.field(AWARENESS_ATTRIBUTE, getAwarenessAttributeHealth());
```

Comment:
This field is the only difference between regular cluster health and this new API

Reply:
Same as above